### PR TITLE
Migrate to create_api_docs.dart

### PR DESCRIPTION
**Pull request by srawlins**
Fixes https://github.com/dart-lang/dartdoc/issues/3481

### CODE REVIEW

**Comment by srawlins** [comment]
CC @gspencergoog I'm having trouble migrating to `create_api_docs.dart`. I tried following the updates made to `docs.sh` in https://github.com/flutter/flutter/commit/ced3e76626b1668b561e43fed9295d3ae99f227b, but I'm still getting:

```none
pub:stdout: Resolving dependencies...
pub:stderr: Because Flutter depends on flutter_test from sdk which doesn't exist (the Flutter SDK is not available), version solving failed.
pub:stderr:
pub:stderr: Flutter users should run `flutter pub get` instead of `dart pub get`.
```

This is being printed by the `runPubProcess` command at line 450 in `create_api_docs.dart`. Any ideas?

**Comment by gspencergoog** [comment]
Looking at this.  I think I may have switched to running `dart pub get` in some places instead of `flutter pub get`.

File: tool/task.dart [hunk 1]
```diff
@@ -314,33 +314,42 @@ Future<Iterable<Map<String, Object?>>> _docFlutter({
...
+      flutterRepo.dartCmd,
       ['pub', 'global', 'deactivate', 'snippets'],
     );
   } on SubprocessException {
     // Ignore failure to deactivate so this works on completely clean bots.
   }
   await flutterRepo.launcher.runStreamed(
-    flutterRepo.cacheDart,
+    flutterRepo.dartCmd,
     ['pub', 'global', 'activate', 'snippets'],
```
Comment (gspencergoog) [inline tool/task.dart:327]: We activate a particular version of snippets (0.4.0 at the moment) inside of the `dev/bots/docs.sh` script. Should you be activating the same one?

**Comment by gspencergoog** [comment]
So, can you try out the changes in https://github.com/flutter/flutter/pull/133493 ?  I modified it to call `flutter pub` instead of `dart pub`, so the Flutter SDK should be available then.

**Comment by srawlins** [comment]
Thanks a million, Greg! I think I can test locally against your PR (maybe?) but I'll wait till it lands to land this change.

**Comment by gspencergoog** [comment]
Okay, cool.  Would you be willing to review that change, or should I find someone else?

**Comment by srawlins** [comment]
Looks good Greg!

**Review by scheglov — APPROVED** [review]

**Pull request merged**